File: 2023/01 - January/06/01/Stack.java
// Stack.java

// Stack Class
public class Stack {
	// Private Attributes
	private int size;
	private Node head;

	// Constructor
	public Stack() {
		this.head = null;
		this.size = 0;
	}

	// Push An Item Onto Stack
	public void push(int item) {
		// Add New Node In The Front
		Node node = new Node(item, head);
		head = node;
		size++;

	}

	// Pop An Item From The Stack
	public int pop() {
		// Remove The Head And Return Data
		int data = -1;
		if (head != null) {
			data = head.getData();
			head = head.getNext();
			size--;
		}
		return data;
	}

	// Returns The Size Of The Stack
	public int size() {
		return size;
	}

	// Prints All Item In The Stack
	public void print() {
		if (size > 0) {
			System.out.print("Stack: ");
			Node current = head;
			while (current != null) {
				System.out.print(current.getData() + " ");
				current = current.getNext();
			}
			System.out.println();
		} else {
			System.out.println("Stack is empty.");
		}
	}
}

File: 2023/03 - March/13/02/Stack.cpp
#include "Stack.h"

// Constructor
Stack :: Stack() {
	// Initialize Instance Variables
	this->top = NULL;
	this->numNodes = 0;
}

// Push A Person Onto The Stack
void  Stack :: push(Person * person) {
	// Create A New Node
	Node * newNode = new Node(person, top);
	// New Node Is The New Top
	this->top = newNode;
	// Increment The Number Of Nodes
	this->numNodes++;
}

// Pop A Person And Returns It
Person * Stack :: pop() {
	if (isEmpty()) {
		return NULL;
	}
	// Get Data Of Top Node
	Person * data = top->getData();
	// Backup Top Reference
	Node * backup = top;
	// Advance Top Node
	this->top = top->getNext();
	// Delete Top Node
	delete backup;
	// Decrement The Number Of Nodes
	numNodes--;
	// Return Data
	return data;
}

// Returns The Size Of The Stack
int Stack :: size() {
	return numNodes;
}
// Return True If The Stack Is Empty, False Otherwise
bool Stack :: isEmpty() {
	return numNodes == 0;
}
// Returns The Top Item On The Stack Without Removing It
Person * Stack :: peek() {
	if (isEmpty()) {
		return NULL;
	}
	return top->getData();
}